### DISCUSSION TITLE
ValueError: all samples are censored

### DISCUSSION THREAD

**Discussion by samarthjmodi** [sebp/scikit-survival#171] [Q&A]
How to deal with this?

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-180-0025a8cdd4f5> in <module>
----> 1 rsf.fit(np.array(x_train), y)

/opt/conda/lib/python3.7/site-packages/sksurv/ensemble/forest.py in fit(self, X, y, sample_weight)
    235         self
    236         """
--> 237         X, event, time = check_arrays_survival(X, y)
    238 
    239         self.n_features_ = X.shape[1]

/opt/conda/lib/python3.7/site-packages/sksurv/util.py in check_arrays_survival(X, y, **kwargs)
    192         Time of event or censoring.
    193     """
--> 194     event, time = check_y_survival(y)
    195     kwargs.setdefault("dtype", numpy.float64)
    196     X = check_array(X, ensure_min_samples=2, **kwargs)

/opt/conda/lib/python3.7/site-packages/sksurv/util.py in check_y_survival(y_or_event, allow_all_censored, *args)
    148 
    149     if not (allow_all_censored or numpy.any(event)):
--> 150         raise ValueError('all samples are censored')
    151 
    152     return_val = [event]

ValueError: all samples are censored

**Comment by sebp** [comment]
The event indicator, part of `y`, is False everywhere, which indicates that all your time points are censored. This is not supported. Unless you made a mistake in constructing `y`, there's no fix for it.
